### DISCUSSION TITLE
Wall faces are not allowed to deform

### DISCUSSION THREAD

**Discussion by sreekarrs** [mdolab/dafoam#493] [General]
Hello

I have been testing shape optimization with different geometry cases and noticed that faces with wall patchType is deforming during the design optimization runs.
Is it possible to allow "wall" faces to deform during the optimization?

If not, is there any alternative method you can suggest to use wall type and allow deformation to that face?

To better understand my issue I attach the following picture where the square hole is the design surface and I expect the result to have a smoother cross-section. But due to the limitation, the square cross-section is being preserved on the top and bottom faces.

![image](https://github.com/mdolab/dafoam/assets/26832867/e37d1b93-2f52-4444-807a-de5e9bca25b5)

Best regards,
Sreekar

**Comment by friedenhe** [comment]
> faces with wall patchType is deforming during the design optimization runs.

I guess you meant "faces with wall patchType is NOT deforming during .."? 

As I understand, you wanted the IDWarp to deform the wall-type patches. This is not supported. I guess the problem you had was caused by an improper FFD setup. From the pic you sent, it seems that you set up an FFD that included only the square hole. However, this is not a recommended way to use FFD. You should generate a bigger FFD that covers all wall patches that are connected. In other words, the FFD should cover both the square hole and the top patch. You then can deform the top patch with FFD, not with IDWarp. Your setup is similar to a wing+fuselage configuration, we can't simply set up a FFD to cover only the wing, we need a bigger FFD that covers both the wing and fuselage. See the FFD pic from here: https://dafoam.github.io/mydoc_tutorials_aero_dpw4.html
